think i added steam support (pls check this is 100% ai lmao)

dxrkinfuser44 opened the pull request:
hi vapid

JoshuaBrest commented:
This code will not work properly. Please try your code before submitting a PR. Lots of the changes don't make sense.

dxrkinfuser44 commented:
This pull request includes several changes to improve the user interface and functionality of multiple views in the Mythic application. Key changes involve refactoring code for better readability and maintainability, updating UI elements, and enhancing user experience with new features.

### Refactoring and Code Simplification:
* [`Mythic/Views/Navigation/Accounts.swift`](diffhunk://#diff-f340ccef70fe6c767a71fb68c2a1a254611829e11249a47f93afcd3421fd7fe0R7-L123): Refactored the `AccountsView` to use a new `AccountRow` component, simplifying the code and reducing redundancy.
* [`Mythic/Views/Navigation/Store.swift`](diffhunk://#diff-634c0acbe6e163e3335d247d8efa514a6a49c6ce2d49507a154f630e7d8e8757L8-R88): Refactored the `StoreView` to use dedicated functions for updating Discord presence, navigation, and refreshing the page.

### UI Improvements:
* [`Mythic/Views/Navigation/Settings.swift`](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L35-R39): Updated various state variables to use default boolean values instead of optionals, and adjusted the display logic for success indicators. [[1]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L35-R39) [[2]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L206-R209) [[3]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L220-R225) [[4]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L249-R256) [[5]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L277-R286) [[6]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L298-R306) [[7]](diffhunk://#diff-788afd51ec157dfca30a9ce878523e6f4d680192f149d0dbd9e63f9f3895cea5L309-R320)

### New Features:
* [`Mythic/Views/Navigation/Bottles.swift`](diffhunk://#diff-d73e689564cd53f5991da52371b4c01d4782f11c4a3efad9464c44035daf56fcR25-L50): Added a toolbar with buttons for adding a bottle and opening the bottles directory. Also introduced a background task to update Discord presence. [[1]](diffhunk://#diff-d73e689564cd53f5991da52371b4c01d4782f11c4a3efad9464c44035daf56fcR25-L50) [[2]](diffhunk://#diff-d73e689564cd53f5991da52371b4c01d4782f11c4a3efad9464c44035daf56fcL60-L66)

### Minor Changes:
* [`Mythic/Views/Navigation/DownloadsViewEvo.swift`](diffhunk://#diff-ee1960f56a86a4c2c3db7ff2278131e64d9b89ff06e8f711e8f12240c77dbf05R28-R34): Added a `LazyVStack` for better performance when displaying a list of download cards.
* Removed redundant or outdated comments from multiple files to clean up the codebase. [[1]](diffhunk://#diff-d73e689564cd53f5991da52371b4c01d4782f11c4a3efad9464c44035daf56fcL10-R11) [[2]](diffhunk://#diff-ee1960f56a86a4c2c3db7ff2278131e64d9b89ff06e8f711e8f12240c77dbf05L8-L14) [[3]](diffhunk://#diff-634c0acbe6e163e3335d247d8efa514a6a49c6ce2d49507a154f630e7d8e8757L8-R88)

JoshuaBrest commented:
Since you asked. I will add comments to the PR.

JoshuaBrest commented on Mythic/Views/Navigation/Accounts.swift:
```diff
@@ -4,111 +4,45 @@
     @State private var isAuthViewPresented: Bool = false
-    @State private var isHoveringOverDestructiveEpicButton: Bool = false
-    @State private var isHoveringOverDestructiveSteamButton: Bool = false
+    @State private var isHoveringOverDestructiveButton: Bool = false
```
You can't use the same variable for both buttons.

JoshuaBrest commented on Mythic/Views/Navigation/Accounts.swift:
```diff
@@ -122,6 +56,61 @@ struct AccountsView: View {
+    }
+}
+
+struct AccountRow: View {
```
If it were three or more rows, maybe, but it's just two. There is no need for this.

JoshuaBrest commented on Mythic/Views/Navigation/Bottles.swift:
```diff
@@ -7,8 +7,8 @@
 
-// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
-// This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+// This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the Licen[...]
```
???

JoshuaBrest commented on Mythic/Views/Navigation/Store.swift:
```diff
@@ -5,103 +5,87 @@
+        WebView(url: url, error: $loadingError, isLoading: $isLoading, canGoBack: $canGoBack, canGoForward: $canGoForward)
+            .navigationTitle("Store")
+            .task(priority: .background) {
+                updateDiscordPresence()
```
This does not need to be a function. It is only used once.

JoshuaBrest commented on Mythic/Views/Navigation/Store.swift:
```diff
@@ -5,103 +5,87 @@
+        }()
+    }
+    
+    private func goBack() {
```
Again, these functions are only used once. They do not need to be functions.

JoshuaBrest commented on Mythic/Views/Navigation/Settings.swift:
```diff
@@ -32,11 +32,11 @@ struct SettingsView: View {
-    @State private var isEngineRemovalSuccessful: Bool?
-    @State private var isCleanupSuccessful: Bool?
-    @State private var isEpicCloudSyncSuccessful: Bool?
+    @State private var isForceQuitSuccessful: Bool = false
```
We need these to be nullable because otherwise, it will show that they failed before the user clicks on them.

JoshuaBrest commented on Mythic/Views/Navigation/Bottles.swift:
```diff
@@ -22,47 +22,58 @@ struct BottlesView: View {
+    }
+}
+
+struct AddBottleButton: View {
```
This is used only once.

JoshuaBrest commented on Mythic/Views/Navigation/Bottles.swift:
```diff
@@ -22,47 +22,58 @@ struct BottlesView: View {
+    }
+}
+
+struct OpenBottlesDirectoryButton: View {
```
This is used only once.
